gene_finder.py

```diff
@@ -108,59 +182,64 @@ def find_all_ORFs_both_strands(dna):
-    'ATGCTACATTCGCAT'
-    """
-    # TODO: implement this
-    pass
-
-
-def longest_ORF_noncoding(dna, num_trials):
-    """ Computes the maximum length of the longest ORF over num_trials shuffles
-        of the specfied DNA sequence
-
-        dna: a DNA sequence
-        num_trials: the number of random shuffles
-        returns: the maximum length longest ORF """
-    # TODO: implement this
-    pass
-
-
-def coding_strand_to_AA(dna):
-    """ Computes the Protein encoded by a sequence of DNA.  This function
-        does not check for start and stop codons (it assumes that the input
-        DNA sequence represents an protein coding region).
-
-        dna: a DNA sequence represented as a string
-        returns: a string containing the sequence of amino acids encoded by the
-                 the input DNA fragment
-
-        >>> coding_strand_to_AA("ATGCGA")
-        'MR'
-        >>> coding_strand_to_AA("ATGCCCGCTTT")
-        'MPA'
-    """
-    # TODO: implement this
-    pass
-
-
-def gene_finder(dna):
-    """ Returns the amino acid sequences that are likely coded by the specified dna
-
-        dna: a DNA sequence
-        returns: a list of all amino acid sequences coded by the sequence dna.
-    """
-    # TODO: implement this
-    pass
+    finallistofORFs = []
+    finallistofORFs.extend(find_all_ORFs(dna))
+    finallistofORFs.extend(find_all_ORFs(get_reverse_complement(dna)))
+    return finallistofORFs
+#
+#
+# def longest_ORF(dna):
+#     """ Finds the longest ORF on both strands of the specified DNA and returns it
+#         as a string
+#     >>> longest_ORF("ATGCGAATGTAGCATCAAA")
+#     'ATGCTACATTCGCAT'
+#     """
+#     # TODO: implement this
+#     pass
+#
+#
+# def longest_ORF_noncoding(dna, num_trials):
+#     """ Computes the maximum length of the longest ORF over num_trials shuffles
+#         of the specfied DNA sequence
+#
+#         dna: a DNA sequence
+#         num_trials: the number of random shuffles
+#         returns: the maximum length longest ORF """
+#     # TODO: implement this
+#     pass
+#
+#
+# def coding_strand_to_AA(dna):
+#     """ Computes the Protein encoded by a sequence of DNA.  This function
+#         does not check for start and stop codons (it assumes that the input
+#         DNA sequence represents an protein coding region).
+#
+#         dna: a DNA sequence represented as a string
+#         returns: a string containing the sequence of amino acids encoded by the
+#                  the input DNA fragment
+#
+#         >>> coding_strand_to_AA("ATGCGA")
+#         'MR'
+#         >>> coding_strand_to_AA("ATGCCCGCTTT")
+#         'MPA'
+#     """
+#     # TODO: implement this
+#     pass
+#
+#
+# def gene_finder(dna):
+#     """ Returns the amino acid sequences that are likely coded by the specified dna
+#
+#         dna: a DNA sequence
+#         returns: a list of all amino acid sequences coded by the sequence dna.
+#     """
+#     # TODO: implement this
+#     pass
```

Comment:
Looks like you forgot to push your code for part2. can you push those changes as well?